Clarify scope to unreliable data only (#37)
quicwg/datagram: draft-ietf-quic-datagram.md

This document defines two new DATAGRAM QUIC frame types, which
carry application data without requiring retransmissions.

> Note that DATAGRAM frames are scoped to unreliable tranmission only. Reliable
> transmission of short datagrams is already supported via STREAM frames.

Reviewer: ```suggestion
Note that DATAGRAM frames are only meant for unreliable transmissions. Reliable
transmission is already supported by QUIC via STREAM frames.
```
